app/src/main/java/com/android/shelfLife/ui/camera/CameraView.kt

```diff
@@ -65,30 +87,110 @@ fun BarcodeScannerScreen(
     }
   }
 
-  // Rest of your BarcodeScannerScreen UI
-  if (permissionGranted) {
-    // Display the camera preview
-    CameraPreviewView(modifier = Modifier.fillMaxSize()) { previewView ->
-      startCamera(context, previewView)
-    }
-  } else {
-    // Optionally, display a message or placeholder
-    Text("Camera permission is required.")
+  Scaffold(
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { selected -> navigationActions.navigateTo(selected) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = Route.SCANNER)
+      }) { paddingValues ->
+        if (permissionGranted) {
+          // Display the camera preview
+          Box(modifier = Modifier.fillMaxSize().padding(paddingValues)) {
+            // Remember the ROI rectangle
+            val roiRectF = remember { mutableStateOf<RectF?>(null) }
+
+            // Camera Preview
+            CameraPreviewView(
+                modifier = Modifier.fillMaxSize(),
+                onBarcodeScanned = { scannedBarcode ->
+                  // Handle the scanned barcode
+                  Log.d("BarcodeScanner", "Scanned barcode: $scannedBarcode")
+                  beep()
+                  // Update ViewModel or navigate as needed
+                  foodFactsViewModel.searchByBarcode(scannedBarcode.toLong())
+                    Toast.makeText(context, "Scanned barcode: $scannedBarcode", Toast.LENGTH_SHORT).show()
+                  //cameraViewModel.onBarcodeScanned(scannedBarcode)
+                },
+                onPreviewViewCreated = { previewView ->
+                  // Do nothing for now
+                },
+                roiRect = roiRectF.value ?: RectF(0f, 0f, 1f, 1f) // Default ROI
+                )
+
+            // Overlay
+            ScannerOverlay { calculatedRoiRectF -> roiRectF.value = calculatedRoiRectF }
+          }
+        }
+      }
+}
+
+@Composable
+fun ScannerOverlay(onRoiCalculated: (RectF) -> Unit) {
+  Canvas(modifier = Modifier.fillMaxSize()) {
+    val canvasWidth = size.width
+    val canvasHeight = size.height
+
+    val rectWidth = canvasWidth * 0.8f
+    val rectHeight = canvasHeight * 0.2f
+    val left = (canvasWidth - rectWidth) / 2f
+    val top = (canvasHeight - rectHeight) / 2f
+
+    // Semi-transparent background
+    drawRect(
+        color = Color(0x40FFFFFF), // 50% opacity black
```

Comment:
The `0.8f` and `0.2f` values for the rectangle's width and height, as well as the color `0x40FFFFFF`, are currently hardcoded. It would be a good idea to extract these into constants to improve readability, allow easier adjustments, and avoid magic numbers in the code. This will also make it simpler to modify these values in the future if needed.